### PR TITLE
[New] Backing Up a Compute Instance to Object Storage

### PR DESCRIPTION
**Pull request by nasanos**
This PR adds a guide titled "Backing Up a Compute Instance to Object Storage". This is a tutorial that walks a use through each step of the process of archiving your files and uploading that archive to Object Storage through the rclone tool.

[Preview](https://deploy-preview-6146--nostalgic-ptolemy-b01ab8.netlify.app/docs/guides/backing-up-compute-instance-to-object-storage/)

### CODE REVIEW

File: docs/guides/platform/object-storage/backing-up-compute-instance-to-object-storage/index.md [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+
+```command
+sudo mv /tmp/backup.tgz /
+cd /
+sudo tar xvpfz backup.tgz -C /
+```
+
+### Backup via rsync
+
+The rsync utility provides robust and versatile options for copying data from one location to another. Here, rsync is used precisely and just to do that, but rsync also includes the noteworthy feature of incremental backups. That feature would allow you to only alter a backup directory relative to changes in the source directory or directories.
```
Comment (wildmanonline) [inline docs/guides/platform/object-storage/backing-up-compute-instance-to-object-storage/index.md:71]: This sentence could generally be improved:

> Here, rsync is used precisely and just to do that, but rsync also includes the noteworthy feature of incremental backups.

**Review by wildmanonline — CHANGES_REQUESTED** [review]
I started a review on this and left a few comments, but I had a question about the purpose of using rsync. If the goal is incremental backups, how does this compare to using the `--listed-incremental` option of the tar command? Perhaps rsync is better, but it creates additional complexity and additional commands. If its possible to only use tar without performance issues or a reduction in functionality over rsync, I'd favor just using tar here.

Also, the sections are positioned as tar vs rsync instead of the real differentiators: Full backups (using tar) and Incremental backups (currently using rsync and tar but perhaps only tar is needed).

**Comment by nasanos** [comment]
@wildmanonline I don't recall whether or not I came across the `--listed-incremental` option at the time. And just now in a few small-scale tests, the tar command seems to work just fine alone, so it would be great if we could approach it as you've suggested.

But that said, I seem to recall my motivation for rsync coming after I ran into limits with tar's efficiency. I'd need to run a full-scale test with it again, but I think that tar struggled or failed with incremental backups on the system scale, due to the size and duration, even on a fresh compute instance.

File: docs/guides/platform/object-storage/backing-up-compute-instance-to-object-storage/index.md [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+
+### Backup via tar
+
+The tar command creates and extracts archives and is useful for compressing collections of files. Both this backup method and the next use tar for this purpose. But the simplest backup approach just uses tar itself, without any other tools.
+
+Here, you execute a `tar` command from the system's root directory (`/`). The command creates an archive of the root directory and stores it as `backup.tgz`. The first `--exclude` option ensures that tar does not attempt to archive the archive itself.
+
+The remaining `--exclude` options exclude the contents of various directories from the backup. You can adjust this as needed, but the options here provide a general guide.
+
+```command
```
[the remark this answers is not in the document]
Comment (nasanos) [reply docs/guides/platform/object-storage/backing-up-compute-instance-to-object-storage/index.md:53]: Looks like this is possible. The root directory is already specified as the source via the `/` at the end of the `tar` command. And, after a little digging, it does look like specific archive file locations can be given after the `-f` option. So, I made this variant on the command which should forgo the need for the `cd` and `mv` commands:

```
sudo tar -vcpzf /tmp/backup.tgz --exclude='backup.tgz' --exclude='dev/*' --exclude='proc/*' --exclude='sys/*' --exclude='tmp/*' --exclude='run/*' --exclude='mnt/*' --exclude='media/*' --exclude='lost+found/*' /
```

**Review by nasanos — COMMENTED** [review]
Reviewed the requested changes, and submitted a new commit with corresponding edits.

**Review by wildmanonline — APPROVED** [review]

**Pull request merged**